### PR TITLE
Add Sol faction tests

### PR DESCRIPTION
**Pull request by kropptrevor**

### CODE REVIEW

**Comment by pgsandstrom** [comment]
Looks good. Out of curiosity, why did you add a bunch of ignorePatterns to eslintconfig? Is it some IDE-integration that you use that was causing issues or something? I have had no problems with it.

File: core/factions/sol.test.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  result: BattleReport,
+  attackerHitChance: number,
+  defenderHitChance: number,
+) {
+  const aHitChance = attackerHitChance
+  const aMissChance = 1 - aHitChance
+  const dHitChance = defenderHitChance
+  const dMissChance = 1 - dHitChance
+  // using closed form for geometric series
+  // P(Muaat missing and Sol hitting) / (1 - P(both missing))
```
Comment (pgsandstrom) [inline core/factions/sol.test.ts:185]: Comment refeers to "muaat" and "sol" but variables are called "attacker" and "defender". Perhaps comments should also be called that?

Comment (kropptrevor) [reply core/factions/sol.test.ts]: Good point. I've updated it now.

**Comment by kropptrevor** [comment]
> Looks good. Out of curiosity, why did you add a bunch of ignorePatterns to eslintconfig? Is it some IDE-integration that you use that was causing issues or something? I have had no problems with it.

When I run `npx eslint "*"`, it'll error on those files. My IDE also happened to do something similar.
Example output:
```
C:\Users\Trevor\repos\ti4calc\tsconfig.tsbuildinfo
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: tsconfig.tsbuildinfo.
The extension for the file (.tsbuildinfo) is non-standard. You should add "parserOptions.extraFileExtensions" to your config
```

**Comment by pgsandstrom** [comment]
I see. There is a script in package.json to lint the project, so you could just runt `npm run lint`. But I guess your way is equally good. Thank you for the PR! :)

**Pull request merged**